### PR TITLE
 refactor: remove deprecated Gtk::StockID used by some Gtk::Image constructors

### PR DESCRIPTION
**Pull request by rodolforg**
A temporary procedure was added in each file that needed the Image
constructor missing in Gtk versions previous 3.24.0.

### CODE REVIEW

**Comment by rodolforg** [comment]
After this PR, the remaining `Gtk::StockID` are from (few) some `Gdk::PixBuf` loaders and from deprecated `Gtk::Action` that I believe @nickleus27  is/was dealing in PR #2589 .

File: synfig-studio/src/gui/canvasview.cpp [hunk 1]
```diff
@@ -871,7 +846,7 @@ CanvasView::create_time_bar()
 	//Setup the Animation Mode Button and the Keyframe Lock button
 	{
 		Gtk::IconSize iconsize=Gtk::IconSize::from_name("synfig-small_icon_16x16");
-		Gtk::Image *icon = manage(new Gtk::Image(Gtk::StockID("synfig-animate_mode_off"), iconsize));
+		Gtk::Image *icon = manage(create_image_from_icon("animate_mode_off_icon", iconsize));
 		animatebutton = Gtk::manage(new class Gtk::ToggleButton());
```
Comment (ice0) [inline synfig-studio/src/gui/canvasview.cpp:850]: I suggest not creating an image widget, but just using the `animatebutton->set_image_from_icon_name(...)` method.

Comment (rodolforg) [reply synfig-studio/src/gui/canvasview.cpp]: Done

File: synfig-studio/src/gui/dialogs/dialog_color.cpp [hunk 2]
```diff
@@ -65,9 +65,9 @@ Dialog_Color::Dialog_Color():
 
 	create_color_edit_widget();
 	create_close_button();
-	create_set_color_button("synfig-set_outline_color", _("Set as Outline"), 0,
+	create_set_color_button("set_outline_color", _("Set as Outline"), 0,
 			sigc::mem_fun(*this, &Dialog_Color::on_set_oc_pressed));
-	create_set_color_button("synfig-set_fill_color", _("Set as Fill"), 1,
+	create_set_color_button("set_fill_color", _("Set as Fill"), 1,
```
Comment (rodolforg) [inline synfig-studio/src/gui/dialogs/dialog_color.cpp:70]: these icons don't follow the filename pattern: do not end with "_icon" :/
Open an issue?

Comment (ice0) [reply synfig-studio/src/gui/dialogs/dialog_color.cpp:70]: I think yes. Better to fix this.

Comment (rodolforg) [reply synfig-studio/src/gui/dialogs/dialog_color.cpp:70]: In this PR or another one?

Comment (rodolforg) [reply synfig-studio/src/gui/dialogs/dialog_color.cpp]: Added here. I can transpose to another PR if you think it is better.

**Comment by rodolforg** [comment]
bad commit before, didn't commit everything

File: synfig-studio/src/gui/canvasview.cpp [hunk 3]
```diff
@@ -1128,11 +1092,8 @@ CanvasView::create_top_toolbar()
 	displaybar->append(*create_tool_separator());
 
 	{ // Refresh button
-		Gtk::Image *icon = Gtk::manage(new Gtk::Image(Gtk::StockID("gtk-refresh"), iconsize));
-		icon->show();
-
-		refreshbutton = Gtk::manage(new class Gtk::ToolButton());
-		refreshbutton->set_icon_widget(*icon);
+		refreshbutton = Gtk::manage(new Gtk::ToolButton());
+		refreshbutton->set_icon_name("gtk-refresh");
```
Comment (ice0) [inline synfig-studio/src/gui/canvasview.cpp:1096]: `gtk-refresh` -> `view-refresh`

File: synfig-studio/src/gui/canvasview.cpp [hunk 4]
```diff
@@ -1268,13 +1220,8 @@ CanvasView::create_top_toolbar()
...
 {
-	Gtk::IconSize iconsize = Gtk::IconSize::from_name("synfig-small_icon_16x16");
-
-	Gtk::Image *icon = Gtk::manage(new Gtk::Image(Gtk::StockID("gtk-stop"), iconsize));
-	icon->show();
-
-	stopbutton = Gtk::manage(new class Gtk::Button());
-	stopbutton->set_image(*icon);
+	stopbutton = Gtk::manage(new Gtk::Button());
+	stopbutton->set_image_from_icon_name("gtk-stop");
```
Comment (ice0) [inline synfig-studio/src/gui/canvasview.cpp:1224]: `process-stop`

File: synfig-studio/src/gui/canvasview.cpp [hunk 5]
```diff
@@ -1991,10 +1922,7 @@ CanvasView::create_tab_label()
 	// Margin is needed to make sure the button is not partially hidden with Adwaita theme
 	closebutton->set_margin_end(4);
 	grid->attach(*closebutton, 1, 0, 1, 1);
-	Gtk::Image* closebutton_image(manage(new Gtk::Image(
-			Gtk::StockID("gtk-close"),
-			Gtk::IconSize::from_name("synfig-small_icon") )));
-	closebutton->add(*closebutton_image);
+	closebutton->set_image_from_icon_name("gtk-close");
```
Comment (ice0) [inline synfig-studio/src/gui/canvasview.cpp:1925]: -> `window-close`

File: synfig-studio/src/gui/canvasview.cpp (hunk 3 above)
Comment (ice0) [inline synfig-studio/src/gui/canvasview.cpp:1096]: `view-refresh`

File: synfig-studio/src/gui/dials/zoomdial.cpp [hunk 6]
```diff
@@ -58,10 +58,10 @@ using namespace studio;
 ZoomDial::ZoomDial(Gtk::IconSize & size):
 	Gtk::Grid()
 {
-	zoom_in = create_icon(size, Gtk::Stock::ZOOM_IN, _("Zoom In"));
-	zoom_out = create_icon(size, Gtk::Stock::ZOOM_OUT, _("Zoom Out"));
-	zoom_fit = create_icon(size, Gtk::Stock::ZOOM_FIT, _("Zoom to Fit"));
-	zoom_norm = create_icon(size, Gtk::Stock::ZOOM_100, _("Zoom to 100%"));
+	zoom_in = create_icon(size, "gtk-zoom-in", _("Zoom In"));
```
Comment (ice0) [inline synfig-studio/src/gui/dials/zoomdial.cpp:61]: `gtk-zoom-in` -> `zoom-in`
`gtk-zoom-out` -> `zoom-out`
`gtk-zoom-100` -> `zoom-original` (https://docs.gtk.org/gtk3/const.STOCK_ZOOM_100.html)
`gtk-zoom-fit` -> `zoom-fit-best` (https://docs.gtk.org/gtk3/const.STOCK_ZOOM_FIT.html)

File: synfig-studio/src/gui/states/state_bline.cpp [hunk 7]
```diff
@@ -1598,58 +1613,41 @@ StateBLine_Context::popup_vertex_menu(synfig::ValueNode_Const::Handle value_node
...
-		item2->show();                                                   \
-		menu.append(*item2);                                             \
-	} while (false)
 
 	bool split_angle = bline_point.get_split_tangent_angle();
 	bool split_radius = bline_point.get_split_tangent_radius();
 	
 	if (split_angle && split_radius)
-		STATE_BLINE_ADD_MENU_ITEM("Merge Tangents", false, false, "gtk-connect");
+		menu.append(*create_split_merge_menu_item(_("Merge Tangents"), false, false, "gtk-connect", value_node));
```
Comment (ice0) [inline synfig-studio/src/gui/states/state_bline.cpp:1623]: `gtk-connect`/`gtk-disconnect` has no replacement :(

https://docs.gtk.org/gtk3/const.STOCK_CONNECT.html

File: synfig-studio/src/gui/canvasview.cpp (hunk 3 above)
Comment (rodolforg) [reply synfig-studio/src/gui/canvasview.cpp:1096]: I just found out orientations about it XD
https://docs.google.com/document/d/1KCVPoYQBqMbDP11tHPpjW6uaEHrvLUmcDPqKAppCY8o/pub
https://docs.google.com/spreadsheets/d/1HavJQRPpMuq-N0GoN1wJR-9KEGXpKy3-NEPpZZkUGJY/pub?output=html

when reading actiongroup here https://gitlab.xfce.org/apps/mousepad/-/issues/58

But I wouldn't have noticed until you pointed. thank you!

Comment (rodolforg) [reply synfig-studio/src/gui/canvasview.cpp:1096]: done

File: synfig-studio/src/gui/canvasview.cpp (hunk 4 above)
Comment (rodolforg) [reply synfig-studio/src/gui/canvasview.cpp:1224]: done

File: synfig-studio/src/gui/canvasview.cpp (hunk 5 above)
Comment (rodolforg) [reply synfig-studio/src/gui/canvasview.cpp:1925]: done

File: synfig-studio/src/gui/canvasview.cpp (hunk 3 above)
Comment (rodolforg) [reply synfig-studio/src/gui/canvasview.cpp:1096]: done

File: synfig-studio/src/gui/dials/zoomdial.cpp (hunk 6 above)
Comment (rodolforg) [reply synfig-studio/src/gui/dials/zoomdial.cpp:61]: done

File: synfig-studio/src/gui/states/state_bline.cpp (hunk 7 above)
Comment (rodolforg) [reply synfig-studio/src/gui/states/state_bline.cpp:1623]: We can create ours for now.
What about those above? They are based on current Bline Tool icon.
merge_split_tangents_icons 0001
merge_split_tangents_icons 0001
merge_split_tangents_icons 0000

Comment (rodolforg) [reply synfig-studio/src/gui/states/state_bline.cpp]: Last commit shows it.
I can delete it if not needed or it is bad

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**